core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycle.scala

```diff
@@ -141,11 +141,13 @@ class BpmBillingProjectLifecycle(
               logger.info(
                 s"Deleting BPM-backed billing project ${projectName}, initiated deletion of landing zone ${landingZoneId} with jobID ${landingZoneResponse.getJobReport.getId}"
               )
+              Future.successful()
           }
         case None =>
           logger.warn(s"Deleting BPM-backed billing project ${projectName}, but no associated landing zone to delete")
+          Future.successful()
       }
-      _ <- billingRepository.getBillingProfileId(projectName).map {
+      result <- billingRepository.getBillingProfileId(projectName).flatMap {
```

Comment:
You don't need to store this in a variable and yield it. The bug was that this was a `Future[Future[Unit]]` rather than a `Future[Unit]`.